jpa/org.eclipse.persistence.jpa/src/main/java/org/eclipse/persistence/internal/jpa/StoredProcedureQueryImpl.java

```diff
@@ -437,13 +437,18 @@ protected StoredProcedureCall getCall() {
 
                 if (parameterType == getCall().INOUT) {
                     field = (DatabaseField) ((Object[]) parameter)[0];
-                } else if (parameterType == getCall().IN || parameterType == getCall().OUT || parameterType == getCall().OUT_CURSOR) {
+                } else if (parameterType == getCall().IN) {
                     field = (DatabaseField) parameter;
-                } else if (parameterType == getCall().LITERAL) {
+                } else if (parameterType == getCall().OUT) {
+                    field = ((OutputParameterForCallableStatement) parameter).getOutputField();
+                } else if (parameterType == getCall().OUT_CURSOR) {
                     if (parameter instanceof OutputParameterForCallableStatement) {
-                        // Case: Oracle OUT_CURSOR after execution.
                         field = ((OutputParameterForCallableStatement) parameter).getOutputField();
+                    } else {
+                        field = (DatabaseField) parameter;
                     }
+                } else if (parameterType == getCall().LITERAL) {
+                    field = (DatabaseField) parameter;
```

Comment:
Is LITERAL supposed to do anything? I mean, the old code flow was: "If LITERAL, only do anything if it's also an `OutputParameterForCallableStatement`". Presumable, the `instanceof` is covered under OUT and OUT_CURSOR now, right? So LITERAL doesnt need to do anything??? Maybe just keep LITERAL as the old implementation and add in OUT/OUT_CURSOR?